LEGO1/mxdirectdraw.cpp

```diff
@@ -0,0 +1,968 @@
+    BOOL fullscreen)
+{
+    HRESULT result;
+    HDC hdc;
+    int i;
+
+    if (g_is_PALETTEINDEXED8)
+    {
+        hdc = GetDC(NULL);
+        GetSystemPaletteEntries(hdc, 0, _countof(m_paletteEntries), m_paletteEntries);
+        ReleaseDC(NULL, hdc);
+    }
+
+    for (i = 0; i < 10; i++)
+    {
+        m_paletteEntries[i].peFlags = 0x80;
+    }
+
+    for (i = 10; i < 142; i++)
+    {
+        m_paletteEntries[i].peFlags = 0x44;
+    }
+
+    for (i = 142; i < 246; i++)
+    {
+        m_paletteEntries[i].peFlags = 0x84;
+    }
+
+    for (i = 246; i < 256; i++)
+    {
+        m_paletteEntries[i].peFlags = 0x80;
+    }
+
+    if (paletteEntryCount != 0)
+    {
+        // actually both of this 'if' statements is not needed(may be it created from for some how)
+        if (paletteEntryCount > 10)
+        {
+            for (i = 10; (i < paletteEntryCount) && (i < 246); i++)
+            {
+                m_paletteEntries[i].peRed = pPaletteEntries[i].peRed;
+                m_paletteEntries[i].peGreen = pPaletteEntries[i].peGreen;
+                m_paletteEntries[i].peBlue = pPaletteEntries[i].peBlue;
+            }
+        }
+    }
+
+    if (m_pPalette != NULL)
+    {
+        result = m_pPalette->SetEntries(0, 0, _countof(m_paletteEntries), m_paletteEntries);
+        if (result != DD_OK)
+        {
+            Error("SetEntries failed", result);
+            return FALSE;
+        }
+    }
+
+    return TRUE;
+}
+
+// OFFSET: LEGO1 0x1009E110
+BOOL MxDirectDraw::TextToTextSurface(
+    const char* text,
+    IDirectDrawSurface* pSurface,
+    SIZE& textSizeOnSurface)
+{
+    HRESULT result;
+    size_t  textLength;
+    HDC hdc;
+    struct tagRECT rc;
+
+    if (pSurface == NULL)
+    {
+        return FALSE;
+    }
+
+    result = pSurface->GetDC(&hdc);
+    if (result != DD_OK)
+    {
+        Error("GetDC for text surface failed", result);
+        return FALSE;
+    }
+
+    textLength = strlen(text) + 1;
+    SelectObject(hdc, m_hFont);
+    SetTextColor(hdc, RGB(255, 255, 0));
+    SetBkColor(hdc, RGB(0, 0, 0));
+    SetBkMode(hdc, OPAQUE);
+    GetTextExtentPoint32(hdc, text, textLength - 1, &textSizeOnSurface);
+    SetRect(&rc, 0, 0, textSizeOnSurface.cx, textSizeOnSurface.cy);
+    ExtTextOut(hdc, 0, 0, ETO_OPAQUE, &rc, text, textLength - 1, NULL);
+    pSurface->ReleaseDC(hdc);
+
+    return TRUE;
+}
+
+// OFFSET: LEGO1 0x1009E210
+BOOL MxDirectDraw::TextToTextSurface1(const char* text)
+{
+    return TextToTextSurface(
+        text,
+        m_pText1Surface,
+        m_text1SizeOnSurface);
+}
+
+// OFFSET: LEGO1 0x1009E230
+BOOL MxDirectDraw::TextToTextSurface2(const char* text)
+{
+    return TextToTextSurface(
+        text,
+        m_pText2Surface,
+        m_text2SizeOnSurface);
+}
+
+// OFFSET: LEGO1 0x1009E020
+void MxDirectDraw::unk1()
+{
+    HRESULT result;
+    byte* line;
+    DDSURFACEDESC ddsd;
+
+    // CRAZY, I dont know what kind of code can produce such of thing
+    //v8 = 2 - !m_bFlipSurfaces;
```

Comment:
```
v8 = m_bFlipSurfaces ? 2 : 0;
```

Reply:
I see you found it out later.

About these comments, they don't add any value to the code.
Perhaps remove these.